src/tests/mod.rs

```diff
@@ -56,7 +56,7 @@ fn shards_to_option_shards<T: Clone>(shards: &[Vec<T>]) -> Vec<Option<Vec<T>>> {
 fn shards_into_option_shards<T>(shards: Vec<Vec<T>>) -> Vec<Option<Vec<T>>> {
     let mut result = Vec::with_capacity(shards.len());
 
-    for v in shards.into_iter() {
+    for v in shards {
```

Comment:
Why do we do this? `shards` is not an array.

Reply:
It's not strictly necessary, just simplifies the code a little.